#########################################
#SRD129 16S - OTU table
#By Mou, KT

#Purpose: Generate an OTU table using files generated by mothur.

#Files needed:
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
#SRD129Flu.outsingletons.abund.taxonomy.csv
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared
#SRD129Flu.outsingletons.abund.subsample.shared.csv

#The output files from mothur needed in R for this section and subsequent sections, aside from fasta files, are text files that can be saved as csv for ease of use in R.

#Load library package
library(tidyverse)

#######################################################################

#To start creating OTU table, edit taxonomy file
#Save "SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy" file generated from mothur as a csv file in a spreadsheet editor.
#Named file as "SRD129Flu.outsingletons.abund.taxonomy.csv"
taxonomy <- read.csv("./data/SRD129Flu.outsingletons.abund.taxonomy.csv") #Import this csv file from working directory using "read.csv" function
taxonomy$Taxonomy <- gsub('*\\(.*?\\) *', '', taxonomy$Taxonomy) #Substitute (100) and variations of that with nothing ('') from the Taxonomy column
taxonomy[1:6,3] #Show column number 3, rows 1 through 6 in 'taxonomy' dataframe
write.csv(taxonomy, file = "SRD129abundsingleton2000taxonomy.csv") #Write 'taxonomy' into a csv file and open the csv file in a spreadsheet editor to remove the size and numbered rows

#Edit subsample.shared file
#Save "SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared" file generated from mothur as a csv file in a spreadsheet editor.
#Named file as "SRD129Flu.outsingletons.abund.subsample.shared.csv"
shared <- read.csv("./data/SRD129Flu.outsingletons.abund.subsample.shared.csv", stringsAsFactors = FALSE)
head(shared) #Check on the first part of 'shared' dataframe
shared[1:6,1]
shared <- t(shared) #Transpose 'shared'
head(shared)
shared[1:6,1]
write.csv(shared, file = 'SRD129abundsingleton2000shared.csv') #Open this csv file in a spreadsheet editor and remove the "V*", "label", and "numOtus" rows

#Read the revised SRD129abundsingleton2000shared.csv file
shared <- read.csv("./data/SRD129abundsingleton2000shared.csv")
colnames(shared) [1] <- "OTU" #Rename first column of 'shared' to "OTU"
taxonomy <- read.csv("./data/SRD129abundsingleton2000taxonomy.csv")
OTUtable <- merge(shared, taxonomy, by.x ="OTU", by.y = "OTU") #Merge 'shared' and 'taxonomy' objects by OTU
head(OTUtable)
nrow(OTUtable) #Count number of rows in 'OTUtable'
ncol(OTUtable) #Count number of columns in 'OTUtable'
write.csv(OTUtable, file= "SRD129abundsingleton2000OTUtable.csv")
#Open this csv file in a spreadsheet editor and remove the first row (numbered rows) and "Size" column

#Check OTU table
OTUtable <-read.csv("./data/SRD129abundsingleton2000OTUtable.csv", stringsAsFactors = FALSE)
head(OTUtable) #Check the first part of 'OTUtable' to make sure it looks ok

#Remove the following samples using a text editor from SRD129abundsingleton2000OTUtable.csv
#and SRD129metadata.csv because these samples were accidentally contaminated during
#DNA extraction process:
#SRD129_P150_N_D28
#SRD129_P151_N_D28
#SRD129_P152_N_D28
#SRD129_P153_N_D28
#SRD129_P154_N_D28
#SRD129_P156_N_D28
#SRD129_P163_N_D28
#SRD129_P164_N_D28
#SRD129_P165_N_D28
#SRD129_P166_N_D28
#SRD129_P167_N_D28
#SRD129_P168_N_D28
