---
title: "SRD129 Flu methods report"
author: "Created by: K. T. Mou"
date: "Updated on: 6Apr2021"
output:
  html_document:
    toc: true
    toc_float: true
    number_sections: true
---
# Purpose
This document provides all R scripts for analyzing and visualizing data from the research paper "Changes in the swine nasal microbiota following influenza A virus challenge in a longitudinal study"

***
# mothur
**Goal**: To use mothur to process paired-end 16S rRNA gene sequences and use output for subsequent analysis in R.

1. Used mothur version 1.40.1, silva release version 132 taxonomy, and fasta files.
  - silva.nr_v132.tax
  - silva.nr_v132.pcr.align (renamed as silva.v4.fasta)
  - *.fastq
2. Run the following commands in same directory where fastq files are. fastq files are from Bioproject PRJNA525911
3. Load mothur program.

<details><summary>SRD129_IAV_Control_mothur_commands.txt</summary>
```{r echo=TRUE, eval=FALSE}
#Create directory of fastq files and file names
make.file(inputdir=<directory path>, type=fastq, prefix=SRD129Flu)
#prefix can be whatever name you want to call the files. We called it SRD129Flu
#Output: 
#SRD129Flu.files

#Check SRD129Flu.files to make sure the text file has the correct three columns (sample name, R1*.fastq, R2*.fastq). If the three columns are not labeled correctly, edit as appropriate using command-line tools, text editor, etc.

#Combine reads and data from all samples
make.contigs(file=SRD129Flu.files, processors=39)
#Output:
#SRD129Flu.trim.contigs.fasta
#SRD129Flu.trim.contigs.qual
#SRD129Flu.scrap.contigs.fasta
#SRD129Flu.scrap.contigs.qual
#SRD129Flu.contigs.report
#SRD129Flu.contigs.groups

#Summarize SRD129Flu.trim.contigs.fasta
summary.seqs(fasta=current)
#Output:
#SRD129Flu.trim.contigs.summary

#Remove sequences with ambiguous bases, anything longer than 275bp, 6 homopolymeric tracts
screen.seqs(fasta=SRD129Flu.trim.contigs.fasta, summary=SRD129Flu.trim.contigs.summary, maxambig=0, maxlength=275, maxhomop=6, group=SRD129Flu.contigs.groups)
#Output:
#SRD129Flu.contigs.pick.groups
#SRD129Flu.trim.contigs.bad.accnos
#SRD129Flu.trim.contigs.good.summary
#SRD129Flu.trim.contigs.good.fasta
#SRD129Flu.contigs.good.groups

#Merge duplicate sequences
unique.seqs(fasta=current)
#Output:
#SRD129Flu.trim.contigs.good.names
#SRD129Flu.trim.contigs.good.unique.fasta

#Generate table with names of unique sequences and names of the groups
count.seqs(name=current, group=current)
#Output: 
#SRD129Flu.trim.contigs.good.count_table

#Summarize SRD129Flu.trim.contigs.good.count_table, SRD129Flu.trim.contigs.good.unique.fasta
summary.seqs(count=current, fasta=current, processors=39)
#Output: 
#SRD129Flu.trim.contigs.good.unique.summary

#Download or import silva full length sequences and taxonomy references from https://mothur.org/wiki/Silva_reference_files
#Unzip tar.gz file and use the silva*.align file for pcr.seqs command
#I used silva version 132, which was the latest version at the time.
pcr.seqs(fasta=silva.seed_v132.align, start=11894, end=25319, keepdots=F)
#Output: 
#silva.nr_v132.pcr.align
#silva.nr_v132.bad.accnos

#Rename to a shorter file name
rename.file(input=silva.seed_v132.pcr.align, new=silva.v4.fasta)

summary.seqs(fasta=silva.v4.fasta)
#Output: silva.v4.summary

#Align sequences to reference sequences
align.seqs(fasta=SRD129Flu.trim.contigs.good.unique.fasta, reference=silva.v4.fasta, flip=T)
#Output:
#SRD129Flu.trim.contigs.good.unique.align
#SRD129Flu.trim.contigs.good.unique.align.report

summary.seqs(fasta=current, count=SRD129Flu.trim.contigs.good.count_table)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.summary

#Keep sequences that start at or before position 1968 and end at or after position 11550
screen.seqs(fasta=SRD129Flu.trim.contigs.good.unique.align, count=SRD129Flu.trim.contigs.good.count_table, start=1968, end=11550, maxhomop=6)
#Output: 
#SRD129Flu.trim.contigs.good.unique.bad.accnos
#SRD129Flu.trim.contigs.good.unique.good.align
#SRD129Flu.trim.contigs.good.good.count_table
#SRD129Flu.trim.contigs.good.pick.count_table

summary.seqs(fasta=current, count=current)
#Output: 
#SRD129Flu.trim.contigs.good.unique.good.summary

#Filter out sequences with overhangs at both ends of the region
filter.seqs(fasta=SRD129Flu.trim.contigs.good.unique.good.align, vertical=T, trump=.)
#Output:
#SRD129Flu.filter
#SRD129Flu.trim.contigs.good.unique.good.filter.fasta

unique.seqs(fasta=current, count=SRD129Flu.trim.contigs.good.good.count_table)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.count_table
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.fasta

#Pre-cluster sequences by allowing up to 2 differences between sequences.
pre.cluster(fasta=current, count=current, diffs=2)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.fasta
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.count_table
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.*.map

#Remove chimeric sequences
chimera.vsearch(fasta=current, count=current, dereplicate=t)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.chimeras
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.accnos

#Remove chimeric sequences from fasta file
remove.seqs(fasta=current, accnos=current)
#Output: 
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta

summary.seqs(fasta=current, count=current)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.summary

#Classify sequences
classify.seqs(fasta=SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta, count=SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.count_table, reference=silva.v4.fasta, taxonomy=silva.nr_v132.tax, cutoff=80)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.taxonomy
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.tax.summary

#Remove undesirable sequences with specific taxon labels
remove.lineage(fasta=current, count=current, taxonomy=current, taxon=Chloroplast-Mitochondria-unknown-Archaea-Eukaryota)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.taxonomy
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table

summary.tax(taxonomy=current, count=current)
#Output:
#SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.tax.summary

#Rename the following files to a simpler name
mv SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta SRD129Flu.outsingletons.fasta 
mv SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.denovo.vsearch.pick.pick.count_table SRD129Flu.outsingletons.count_table

#Take out all the sequences that occur once
split.abund(fasta=SRD129Flu.outsingletons.fasta, count=SRD129Flu.outsingletons.count_table, cutoff=1, accnos=true)
#Output:
#SRD129Flu.outsingletons.rare.count_table
#SRD129Flu.outsingletons.abund.count_table
#rare.accnos
#abund.accnos
#SRD129Flu.outsingletons.rare.fasta
#SRD129Flu.outsingletons.abund.fasta

#Calculate uncorrected pairwise distances between aligned DNA sequences
dist.seqs(fasta=SRD129Flu.outsingletons.abund.fasta, cutoff=0.03)
#Output:
#SRD129Flu.outsingletons.abund.dist

#Assign sequences to OTUs
cluster(column=current, count=SRD129Flu.outsingletons.abund.count_table)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.list
#SRD129Flu.outsingletons.abund.opti_mcc.steps
#SRD129Flu.outsingletons.abund.opti_mcc.sensspec

#Determine number of sequences in each OTU from each sample
make.shared(list=current, count=SRD129Flu.outsingletons.abund.count_table, label=0.03)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.shared

#Identify taxonomies for each OTU
classify.otu(list=SRD129Flu.outsingletons.abund.opti_mcc.list, count=SRD129Flu.outsingletons.abund.count_table, taxonomy=SRD129Flu.trim.contigs.good.unique.good.filter.unique.precluster.pick.nr_v132.wang.pick.taxonomy, label=0.03)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.tax.summary

#Files of importance
#count=SRD129Flu.outsingletons.abund.count_table
#shared=SRD129Flu.outsingletons.abund.opti_mcc.shared 
#constaxonomy=SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy

#Determine number of sequences in each sample
count.groups(shared=SRD129Flu.outsingletons.abund.opti_mcc.shared)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.count.summary

#Normalize data
sub.sample(shared=SRD129Flu.outsingletons.abund.opti_mcc.shared, size=2000)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared

#See number of sequences in each sample after subsampling as well as number of sequences total
count.groups(shared=current)
#Output:
#SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.count.summary

#Determine final number of sequences
summary.seqs(fasta=SRD129Flu.outsingletons.abund.fasta)
#Output:
#SRD129Flu.outsingletons.abund.summary
```

## Check quality of fastq sequences
**Goal**: Check fastq sequence quality using fastQC and multiQC, and work within a new directory
<details><summary>fastqc and multiqc scripts</summary>
```{r echo=TRUE, eval=FALSE}
#Download fastqc to your computer and enter in the terminal the following commands:
fastqc
mkdir fastqc
fastqc *.fastq -o fastqc

#Output:
#*fastqc.zip
#*fastqc.html

#Copy all nasal fastqc files to the new directory. MultiQC for nasal wash
multiqc .
#Output:
SRD129Flu_nasal_multiqc_data
SRD129Flu_nasal_multiqc.html
```
</details>

## Creating OTU table
**Goal**: Create OTU table with R using specific files generated from mothur

1. Files needed:
  - SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
  - SRD129Flu.outsingletons.abund.taxonomy.csv
  - SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared
  - SRD129Flu.outsingletons.abund.subsample.shared.csv
2. The output files from mothur needed in R for this section and subsequent sections, aside from fasta files, are text files that can be saved as csv for ease of use in R.

<details><summary>SRD129_OTUtable_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library package
library(tidyverse)

######################

#To start creating OTU table, edit taxonomy file
#Save "SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy" file generated from mothur as a csv file in a spreadsheet editor. 
#Named file as "SRD129Flu.outsingletons.abund.taxonomy.csv"
taxonomy <- read.csv("./data/SRD129Flu.outsingletons.abund.taxonomy.csv") #Import this csv file from working directory using "read.csv" function
taxonomy$Taxonomy <- gsub('*\\(.*?\\) *', '', taxonomy$Taxonomy) #Substitute (100) and variations of that with nothing ('') from the Taxonomy column
taxonomy[1:6,3] #Show column number 3, rows 1 through 6 in 'taxonomy' dataframe
write.csv(taxonomy, file = "SRD129abundsingleton2000taxonomy.csv") #Write 'taxonomy' into a csv file and open the csv file in a spreadsheet editor to remove the size and numbered rows

#Edit subsample.shared file
#Save "SRD129Flu.outsingletons.abund.opti_mcc.0.03.subsample.shared" file generated from mothur as a csv file in a spreadsheet editor. 
#Named file as "SRD129Flu.outsingletons.abund.subsample.shared.csv"
shared <- read.csv("./data/SRD129Flu.outsingletons.abund.subsample.shared.csv", stringsAsFactors = FALSE)
head(shared) #Check on the first part of 'shared' dataframe
shared[1:6,1]
shared <- t(shared) #Transpose 'shared'
head(shared)
shared[1:6,1]
write.csv(shared, file = 'SRD129abundsingleton2000shared.csv') #Open this csv file in a spreadsheet editor and remove the "V*", "label", and "numOtus" rows

#Read the revised SRD129abundsingleton2000shared.csv file
shared <- read.csv("./data/SRD129abundsingleton2000shared.csv")
colnames(shared) [1] <- "OTU" #Rename first column of 'shared' to "OTU"
taxonomy <- read.csv("./data/SRD129abundsingleton2000taxonomy.csv")
OTUtable <- merge(shared, taxonomy, by.x ="OTU", by.y = "OTU") #Merge 'shared' and 'taxonomy' objects by OTU
head(OTUtable)
nrow(OTUtable) #Count number of rows in 'OTUtable'
ncol(OTUtable) #Count number of columns in 'OTUtable'
write.csv(OTUtable, file= "SRD129abundsingleton2000OTUtable.csv") 
#Open this csv file in a spreadsheet editor and remove the first row (numbered rows) and "Size" column

#Check OTU table
OTUtable <-read.csv("./data/SRD129abundsingleton2000OTUtable.csv", stringsAsFactors = FALSE)
head(OTUtable) #Check the first part of 'OTUtable' to make sure it looks ok

#Remove the following samples using a text editor from SRD129abundsingleton2000OTUtable.csv 
#and SRD129metadata.csv because these samples were accidentally contaminated during 
#DNA extraction process: 
#SRD129_P150_N_D28 
#SRD129_P151_N_D28 
#SRD129_P152_N_D28 
#SRD129_P153_N_D28 
#SRD129_P154_N_D28 
#SRD129_P156_N_D28 
#SRD129_P163_N_D28 
#SRD129_P164_N_D28 
#SRD129_P165_N_D28 
#SRD129_P166_N_D28 
#SRD129_P167_N_D28 
#SRD129_P168_N_D28 
```
</details>

***
# phyloseq
**Goal**: Create phyloseq objects to be used to calculate alpha and beta diversity measures for nasal samples. This section will also use the adonis function to determine the effect of time and treatment on the community structure of nasal microbiota.

1. Files needed:
  - SRD129abundsingleton2000OTUtable.csv
  - SRD129metadata.csv

<details><summary>SRD129_phyloseq_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Install any Bioconductor packages
#bioc_pkgs <- c("vegdist")
#BiocManager::install(bioc_pkgs)

#Load library packages
library(vegan)
library(ggplot2)
library(dplyr)
library(tidyr)
library(tidyverse)
library(phyloseq)
library(vegdist)

######################

#Read files
otu <- read.csv("./data/SRD129abundsingleton2000OTUtable.csv", row.names=1) #Set column 1 as row names
meta <-read.csv("./data/SRD129metadata.csv")

dim(otu) #Check dimensions of 'otu'
head(otu[,1:5]) #Check the first part of 'otu' table
head(otu[,185:187]) #Check last part of 'otu' table
dim(meta) #Determine the dimensions of 'meta' dataframe. It has 186 rows and 5 columns
head(meta[,1:5]) #Check first part of 'meta' table

#Remove taxonomy from 'otu'
tax <- otu[,(186:187)] #Remove column 186 taxonomy and 187 to copy the row names from 'otu' to 'tax'
head(tax)
colnames(tax)[1] <- "delete" #Rename column 1 of 'tax' (formerly column 186) as "delete" which will be deleted later
head(tax)

#Modify 'otu' with only OTU count data
otu <- otu[,-187] #Remove column 187 taxonomy in 'otu' to have only OTU data
head(otu[,180:186]) 
dim(otu) #Dimensions of 'otu' show 1299 rows 186 columns

#Transpose 'otu' to match format of 'meta'
otu.trans <- t(otu) 
#Now rownames in 'otu.trans' are sample names, columns are OTUs
head(otu.trans[,1:5])
head(meta) #Row names are numbered, but we want sample names as row names
rownames(meta) <- meta$Sample #Set names in "Sample" as rownames in 'meta'
head(meta)
class(meta) #The type of class that 'meta' is is dataframe
class(otu) #dataframe

#Merge 'otu' and 'meta' data frames
otu.meta <- merge(meta, otu.trans, by.x=0, by.y=0) #Merge by names of the columns that are common to both x and y (columns with common names between the two data sets)
#by.x=0 means match by rownames in 'meta'; y=0 means match by rownames in 'otu.trans'
head(otu.meta[,1:10])
class(otu.meta) #Check class type of 'otu.meta'. It should be a dataframe.

#Added an "All" column (combines 'Treatment' and 'Day' values) in new 'otu.meta2' dataframe
otu.meta2<- cbind(otu.meta) #Make second copy of 'otu.meta' to use to include an "All" column
otu.meta2$All <- with(otu.meta2, paste0(Day, sep=" ", Treatment)) #Combine "Day" and "Treatment" columns into an "All" column
head(otu.meta2) #Check first part of otu.meta2
dim(otu.meta2) #Check dimensions of 'otu.meta2' dataframe
head(otu.meta2[,1300:1305]) #Check the first part of the end of 'otu.meta2' dataframe
rownames(otu.meta2) <- otu.meta2$Row.names #Set "Row.names" as rownames
otu.meta2 <- otu.meta2[,-1]
head(otu.meta2[,1:10])
dim(otu.meta2) #186 1304
head(otu.meta2[,1300:1304])
otu.meta2<- otu.meta2[,c(1:4,1304,5:1303)] #Reorder columns to have "All" column after "Treatment" column
head(otu.meta2[,1:10]) #Check the first part of the beginning of 'otu.meta2' dataframe
head(otu.meta2[,1300:1303])
write.csv(otu.meta2, file="SRD129abundsingleton2000.otu.meta.csv")

#Creating phyloseq objects for nasal samples

#Pull out metadata from 'otu.meta2' dataframe
head(otu.meta2[,1:10])
dim(otu.meta2) #186 1643
otu.meta3 <- otu.meta2[,1:5] #Take columns 1-5 to make 'otu.meta3'
head(otu.meta3)
dim(otu.meta3) #186 5

#Create SAM metadata table phyloseq object
SAM = sample_data(otu.meta3, errorIfNULL = TRUE)
head(SAM)
dim(SAM) #186 5

#Pull out OTU data from 'otu.meta2' dataframe
head(otu.meta2[,1:10])
dim(otu.meta2) #186 1304
head(otu.meta2[,1300:1304])
otu.meta4 <- otu.meta2[,c(6:1304)] #Select "OTU" columns to create 'otu.meta4' dataframe
head(otu.meta4[,1:10])
dim(otu.meta4) #186 1299
otu.meta4.trans <- t(otu.meta4) #Transpose 'otu.meta4' to have OTUs as rownames, sample names as column names
head(otu.meta4.trans[,1:10])
dim(otu.meta4.trans) #1299  186
head(otu.meta4.trans[,180:186])

#Merge 'tax' back into 'otu.meta4.trans' for correct format and taxons
head(tax)
otu.tax <- merge(otu.meta4.trans, tax, by=0) #Merge by rownames aka OTU rownames
dim(otu.tax) #1299  189
head(otu.tax[,185:189])
head(otu.tax[,1:5])
row.names(otu.tax) <- otu.tax[,1] #Set first row as rownames
head(otu.tax[,1:5])
otu.tax <- otu.tax[,-1] #Remove first row, extraneous OTU column
head(otu.tax[,1:5])
dim(otu.tax) #1299 188
head(otu.tax[185:188])

#Split 'otu.tax.nw2' again
dim(otu.tax) #1299 188
head(otu.tax[185:188])
otu.notax <- otu.tax[,1:186] #Take rows 1-186 to make new dataframe 'otu.notax' (187 is delete column, 188 is taxonomy column)
head(otu.notax[,1:5])
head(otu.notax[,184:186])
dim(otu.notax) #1299  186
class(otu.notax) #dataframe
otu.notax <- as.matrix(otu.notax) #Turn 'otu.notax' into a matrix class
class(otu.notax) #matrix

#Create OTU table phyloseq object
OTU = otu_table(otu.notax, taxa_are_rows = TRUE)
head(OTU)
dim(OTU) #1299  186
class(OTU)
dim(otu.tax) #1299  188
head(otu.tax[,186:188])
tax.levels <- separate(data = otu.tax, 
                       col = Taxonomy, 
                       into=c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";")
#Separate Taxonomy column into 7 separate columns labeled "Kingdom", "Phylum", "Class", etc.
head(tax.levels) #Notice that "Species" column is blank
dim(tax.levels) #1299  194
head(tax.levels[,186:194])
head(tax.levels[,188:193])
tax.only <- tax.levels[,188:193] #Keep only taxonomy columns from "Kingdom" up to "Genus"
head(tax.only)
dim(tax.only) #1299 6
class(tax.only) #data.frame
tax.m <- as.matrix(tax.only)
class(tax.m) #matrix
head(tax.m)

#Create TAX taxonomy table phyloseq object
TAX = tax_table(tax.m)
head(TAX)
dim(TAX) #1299 6
head(TAX)
class(TAX)

#Create phyloseq object 'phyloseqFlu' containing taxonomy, metadata, and OTU table
phyloseqFlu <- phyloseq(OTU, SAM, TAX)
phyloseqFlu #view phyloseq object
#phyloseq-class experiment-level object
#otu_table()   OTU Table:         [ 1299 taxa and 196 samples ]
#sample_data() Sample Data:       [ 196 samples by 5 sample variables ]
#tax_table()   Taxonomy Table:    [ 1299 taxa by 6 taxonomic ranks ]

save(phyloseqFlu, file="SRD129Flu.phyloseq.RData")


#Run adonis function to determine effect of time and treatment on structure of nasal microbiota

adonis.Flu <- as(sample_data(phyloseqFlu), "data.frame")
class(adonis.Flu) #data.frame

#distance function to run distance calculations
dist.Flu <- distance(phyloseqFlu, method="bray") #Distance calculation using Bray-Curtis
set.seed(1) #Use set.seed function when running simulations to ensure all results are reproducible
full.Flu <- adonis(dist.Flu~Day*Treatment, data=adonis.Flu, permutations=9999)
full.Flu #Display results
#Output:
#Call:
#Call:
#  adonis(formula = dist.Flu ~ Day * Treatment, data = adonis.Flu,      permutations = 9999) 

#Permutation: free
#Number of permutations: 9999

#Terms added sequentially (first to last)

#                   Df    SumsOfSqs MeanSqs   F.Model   R2        Pr(>F)    
#  Day              9     8.2853    0.92059   12.8444   0.37628   1e-04 ***
#  Treatment        1     0.3890    0.38903   5.4279    0.01767   1e-04 ***
#  Day:Treatment    9     1.4470    0.16077   2.2432    0.06571   1e-04 ***
#  Residuals        166   11.8977   0.07167             0.54034           
#  Total            185   22.0190                       1.00000           
#---
#  Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
    
###Day had largest effect on variation
    
#switched Treatment and Day order, obtained same conclusions: Day had the largest effect on variation
full.Flu_2 <- adonis(dist.Flu~Treatment*Day, data=adonis.Flu, permutations=9999)
full.Flu_2
#Output:
#Call:
#  adonis(formula = dist.Flu ~ Treatment * Day, data = adonis.Flu,      permutations = 9999) 

#Permutation: free
#Number of permutations: 9999

#Terms added sequentially (first to last)

#                   Df    SumsOfSqs MeanSqs   F.Model   R2        Pr(>F)    
#  Treatment        1     0.4068    0.40676   5.6752    0.01847   2e-04 ***
#  Day              9     8.2676    0.91862   12.8169   0.37548   1e-04 ***
#  Treatment:Day    9     1.4470    0.16077   2.2432    0.06571   1e-04 ***
#  Residuals        166   11.8977   0.07167             0.54034           
#  Total            185   22.0190                       1.00000           
#---
#  Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
    


#If distance function is giving an error message below:
#"Error: x should be a data.frame, data.table, tbl, tbl_df, array, or matrix."
#Use vegdist function from vegan package to run distance calculations instead of the distance function 
#(original "distance" function that was used below is no longer available) and use those calculations to run through adonis test
#vegdist requires that phyloseq object's OTU table has OTUs listed in the columns and sample names listed in rows. 
#Also, remove any OTUs with taxa_sums = 0 or non-numeric values. For example, this command can help remove OTUs with taxa_sums = 0: 
#OTU <- prune_taxa(taxa_sums(<yourOTUtable>) > 0, <yourOTUtable>)
#If you create a separate phyloseq object with this specific OTU table setup, 
#you should be able to run the vegdist function without any errors and use the output to run through adonis function

head(otu.meta4) #Sample names are listed in rows and OTUs are listed in columns in 'otu.meta4'
OTU.2 = otu_table(otu.meta4, taxa_are_rows = TRUE)
OTU.2.distance <- vegdist(OTU.2, method = "bray")
OTU.2.distance.full <- adonis(OTU.2.distance~Day*Treatment, data = adonis.Flu, permutations=9999)
OTU.2.distance.full
#Output is the same as full.Flu:
#Call:
#adonis(formula = OTU.2.distance ~ Day * Treatment, data = adonis.Flu,      permutations = 9999) 

#Permutation: free
#Number of permutations: 9999

#Terms added sequentially (first to last)

#                   Df    SumsOfSqs MeanSqs   F.Model   R2        Pr(>F)    
#  Day              9     8.2853    0.92059   12.8444   0.37628   1e-04 ***
#  Treatment        1     0.3890    0.38903   5.4279    0.01767   1e-04 ***
#  Day:Treatment    9     1.4470    0.16077   2.2432    0.06571   1e-04 ***
#  Residuals      166     11.8977   0.07167             0.54034           
#  Total          185     22.0190                       1.00000           
#---
#  Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```
</details>

***
# Beta diversity
**Goal**: Generate non-metric multidimensional scaling ordination based on Bray-Curtis dissimilarities to create NMDS plots, and runs pairwise.adonis function to identify any significant differences in bacterial composition between treatment groups on a given day.

1. Files needed:
  - phyloseqFlu object from previous step

<details><summary>SRD129_alpha_beta_diversity_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages
library(vegan)
library(ggplot2)
library(dplyr)
library(tidyr)
library(tidyverse)
library(phyloseq)
library(scales)
library(RColorBrewer)
library(philentropy)
library(cowplot)
library('wesanderson')

#######################
    
#Load this function from the funfuns R package (https://github.com/Jtrachsel/funfuns)
NMDS_ellipse <- function(metadata, OTU_table, grouping_set,
                         distance_method = 'bray',
                         rand_seed = 77777,
                         MDS_trymax = 1000,
                         autotransform = FALSE,
                         wascores = TRUE,
                         expand = FALSE){
  require(vegan)
  require(tidyr)
  
  if (grouping_set %in% colnames(metadata)){
    if (all(rownames(metadata) == rownames(OTU_table))){
      
      set.seed(rand_seed)
      generic_MDS <- metaMDS(OTU_table, k = 2,
                             trymax = MDS_trymax,
                             autotransform = autotransform,
                             distance = distance_method,
                             wascores = wascores,
                             expand = expand)
      
      stress <- generic_MDS$stress
      nmds_points <- as.data.frame(generic_MDS$points)
      metadata <- metadata[match(rownames(generic_MDS$points), rownames(metadata)),]
      metadata <- as.data.frame(metadata) # weird things were happening when a grouped tibble was used as metadata...
      metanmds <- cbind(metadata, nmds_points)
      # browser()
      nmds.mean <- aggregate(metanmds[,grep("MDS", colnames(metanmds))], list(group=metanmds[[grouping_set]]), mean)
      metanmds[[grouping_set]] <- factor(metanmds[[grouping_set]]) # this 'set' needs to be passed in from function
      
      #check to make sure at least 3 obs for each grouping_set
      
      numobs <- metanmds %>% group_by(!!grouping_set) %>% summarise(n=n())
      if (min(numobs$n) >= 3){
        ord <- ordiellipse(generic_MDS, metanmds[[grouping_set]], label = TRUE, conf = .95, kind = 'se', draw = 'none')
        
        df_ell <- data.frame()
        for (d in levels(metanmds[[grouping_set]])){
          df_ell <- rbind(df_ell, cbind(as.data.frame(with(metanmds[metanmds[[grouping_set]] == d,],
                                                           vegan:::veganCovEllipse(ord[[d]]$cov, ord[[d]]$center, ord[[d]]$scale))),group=d))
        }
        
        
        
        # this loop assigns the group centroid X coordinates to each sample, there is probably a better way...
        
        metanmds$centroidX <- NA
        metanmds$centroidY <- NA
        
        
        for (level in levels(metanmds[[grouping_set]])){
          metanmds[metanmds[[grouping_set]] == level,]$centroidX <- nmds.mean$MDS1[nmds.mean$group == level]
          metanmds[metanmds[[grouping_set]] == level,]$centroidY <- nmds.mean$MDS2[nmds.mean$group == level]
          
          
        }
        print(paste('Ordination stress:', stress, sep = ' '))
        return(list(metanmds, df_ell, generic_MDS))
        
      } else {
        warning('One of your groups in "grouping_set" has less than 3 observations, cannot generate elipses')
        df_ell <- data.frame()
        return(list(metanmds, df_ell, generic_MDS))}
      
    } else {
      stop('The rownames for your OTU table and metadata do not match.')
    }
    
  }else {
    stop('The grouping set column you have provided in not in your metadata.')
  }
  
  
  
}


#Load the following pairwise.adonis function, taken from https://www.researchgate.net/post/How_can_I_do_PerMANOVA_pairwise_contrasts_in_R
pairwise.adonis <- function(x,factors, sim.function = 'vegdist', sim.method = 'bray', p.adjust.m ='bonferroni')
{
  library(vegan)
  
  co = combn(unique(as.character(factors)),2)
  pairs = c()
  F.Model =c()
  R2 = c()
  p.value = c()
  
  
  for(elem in 1:ncol(co)){
    if(sim.function == 'daisy'){
      library(cluster); x1 = daisy(x[factors %in% c(co[1,elem],co[2,elem]),],metric=sim.method)
    } else{x1 = vegdist(x[factors %in% c(co[1,elem],co[2,elem]),],method=sim.method)}
    
    ad = adonis(x1 ~ factors[factors %in% c(co[1,elem],co[2,elem])] );
    pairs = c(pairs,paste(co[1,elem],'vs',co[2,elem]));
    F.Model =c(F.Model,ad$aov.tab[1,4]);
    R2 = c(R2,ad$aov.tab[1,5]);
    p.value = c(p.value,ad$aov.tab[1,6])
  }
  p.adjusted = p.adjust(p.value,method=p.adjust.m)
  sig = c(rep('',length(p.adjusted)))
  sig[p.adjusted <= 0.05] <-'.'
  sig[p.adjusted <= 0.01] <-'*'
  sig[p.adjusted <= 0.001] <-'**'
  sig[p.adjusted <= 0.0001] <-'***'
  
  pairw.res = data.frame(pairs,F.Model,R2,p.value,p.adjusted,sig)
  print("Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1")
  return(pairw.res)
  
} 

###################3
    
#Setting up 'phyloseqFlu' into dataframes for NMDS calculation
flu.sam <- data.frame(phyloseqFlu@sam_data) #Make 'phyloseqFlu sam_data' into dataframe
flu.otu <- data.frame(t(phyloseqFlu@otu_table)) #Make 'phyloseqFlu otu_table' into dataframe
class(flu.sam) #data.frame
rownames(flu.sam) == row.names(flu.otu) #For rows with sums greater than 1 in 'flu.otu', move rows and their respective sum values into "numOTUs" column in 'flu.sam'
flu.sam$numOTUS <- rowSums(flu.otu > 1)
head(flu.sam)

#NMDS calculation
flu.otu[1:10,1:10]
flu.NMDS <- NMDS_ellipse(flu.sam, flu.otu, grouping_set = 'All')
#Output:
#Result: [1] "Ordination stress: 0.168876156130503"
    
#Separate meta data and ellipse data to two lists to make NMDS plot
flu.NMDS.2 <- flu.NMDS[[1]] 
#'flu.NMDS.2' has meta data + MDS calculations. Select this 1st list of 'flu.NMDS' using double brackets
flu.df_ell.2 <- flu.NMDS[[2]]
#'flu.df_ell.2' is accessing 2nd list from 'flu.NMDS' that has ellipse calculations
#Need two separate lists for making NMDS plot
flu.df_ell.2$group #20 levels
head(flu.df_ell.2)

#Create "Day" and "Treatment" columns within 'nw.df_ell' for faceting purposes
flu.df_ell.2$Day <- sub(' [A-Za-z]+', '\\1', flu.df_ell.2$group) #Created "Day" column, '\\1' returns the first part of the regular expression D[A-z0-9]+ from 'flu.df_ell.2$group'
flu.df_ell.2$Treatment <- sub('D[A-z0-9]+ ', '\\2', flu.df_ell.2$group) #Create "Treatment" column, '\\2' returns the second part of the sub expression ([A-Za-z]+) from 'flu.df_ell.2$group'
head(flu.df_ell.2)

#Restructure level order for 'nw.metanmds' and 'nw.df_ell'
flu.NMDS.2$Day = factor(flu.NMDS.2$Day, levels=c("D0", "D1", "D3", "D7", "D10", "D14", "D21", "D28", "D36", "D42"))
flu.df_ell.2$Day = factor(flu.df_ell.2$Day, levels=c("D0", "D1", "D3", "D7", "D10", "D14", "D21", "D28", "D36", "D42"))
levels(flu.df_ell.2$Day) # [1] "D0"  "D1"  "D3"  "D7"  "D10" "D14" "D21" "D28" "D36" "D42"
levels(flu.NMDS.2$Day) # [1] "D0"  "D1"  "D3"  "D7"  "D10" "D14" "D21" "D28" "D36" "D42"
    
#Creating plot from NMDS calculations
FluNMDSplot <- ggplot(data=flu.NMDS.2, aes(x=MDS1, y=MDS2, color=Treatment)) + geom_point() + 
  geom_segment(aes(x=MDS1, xend=centroidX, y=MDS2, yend=centroidY), alpha = 0.5) + 
  geom_path(data=flu.df_ell.2, aes(x=NMDS1, y=NMDS2, color=Treatment, group=group)) + 
  facet_wrap(~Day, scales = 'free') +
  #scale_color_brewer(palette="Dark2") +
  theme_gray(base_size = 10) +
  theme(strip.text.x = element_text(size=15)) +
  labs(caption = 'Ordination stress = 0.169', color="Treatment group")
FluNMDSplot
    
#Save 'FluNMDSplot' as a .tiff for publication, 500dpi
ggsave("Figure_1.tiff", plot=FluNMDSplot, width = 10, height = 6, dpi = 500, units =c("in"))
    
#Using pairwise.adonis function
flu.adon <- pairwise.adonis(flu.otu, flu.sam$All, sim.method = 'bray', p.adjust.m = 'bonferroni') 
#Run pairwise.adonis on 'flu.otu' OTU table and "All" column of 'flu.sam' dataframe
flu.adon$pairs #List all comparisons in the "pairs" column of 'flu.adon'
goodcomps.flu.adon <- c(grep('D0 [A-Za-z]+ vs D0 [A-Za-z]+', flu.adon$pairs),
                        grep('D1 [A-Za-z]+ vs D1 [A-Za-z]+', flu.adon$pairs),
                        grep('D3 [A-Za-z]+ vs D3 [A-Za-z]+', flu.adon$pairs),
                        grep('D7 [A-Za-z]+ vs D7 [A-Za-z]+', flu.adon$pairs),
                        grep('D10 [A-Za-z]+ vs D10 [A-Za-z]+', flu.adon$pairs),
                        grep('D14 [A-Za-z]+ vs D14 [A-Za-z]+', flu.adon$pairs),
                        grep('D21 [A-Za-z]+ vs D21 [A-Za-z]+', flu.adon$pairs),
                        grep('D28 [A-Za-z]+ vs D28 [A-Za-z]+', flu.adon$pairs),
                        grep('D36 [A-Za-z]+ vs D36 [A-Za-z]+', flu.adon$pairs),
                        grep('D42 [A-Za-z]+ vs D42 [A-Za-z]+', flu.adon$pairs))
# "[A-Za-z]" matches all capital and lowercase letters
# "+" matches a whole word and not just one letter (if you didn't have "+", then it would match by one letter)
# "c" creates the vector, lumps all pairs of specific groups of interest together
# You want to make a vector to combine all the pairwise comparisons you're interested in (same day, different treatment group)
goodcomps.flu.adon
goodcomps.flu.adon.2 <- flu.adon[goodcomps.flu.adon,] #Rename 'goodcomps.flu.adon' vector to 'goodcomps.flu.adon.2'
goodcomps.flu.adon.2
goodcomps.flu.adon.2$p.adjusted <- p.adjust(goodcomps.flu.adon.2$p.value, method = 'fdr') #"p.adjust" function returns a set of p-values adjusted with "fdr" method
goodcomps.flu.adon.2$p.adjusted2 <- round(goodcomps.flu.adon.2$p.adjusted, 3) #Round p-values to 3 decimal points and list in new "p.adjusted2" column
goodcomps.flu.adon.2$p.adjusted2[goodcomps.flu.adon.2$p.adjusted2 > 0.05] <- NA #For all p-values greater than 0.05, replace with "NA"
write.csv(goodcomps.flu.adon.2, file='Flu.pairwisecomparisons.csv', row.names=TRUE)
```
</details>

***
# Alpha diversity
**Goal**: Calculate alpha diversity metrics (Shannon, Inverse Simpson) that are plotted as box and whisker plots, and uses wilcoxon rank sum test to assess any significant differences in diversity between treatment groups on a given day.

1. Files needed:
  - phyloseqFlu object from phyloseq step

<details><summary>SRD129_alpha_beta_diversity_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages
library(vegan)
library(ggplot2)
library(dplyr)
library(tidyr)
library(tidyverse)
library(phyloseq)
library(scales)
library(RColorBrewer)
library(philentropy)
library(cowplot)
library('wesanderson')

###########################

#Calculating alpha diversity metrics: Shannon, Inverse Simpson
flu.sam$shannon <- diversity(otu.meta4) #"diversity" is a vegan function. The default index is set at "shannon". I added a shannon index column in 'flu.sam'
flu.sam$invsimpson <- diversity(otu.meta4,index = 'invsimpson') #We used 'invsimpson' since it is easier to interpret than Simpson values and won't need to "inverse" the Simpson values to understand (With Simpson values, the lower the number, the higher the diversity)
flu.sam$Day = factor(flu.sam$Day, levels=c("D0", "D1", "D3", "D7", "D10", "D14", "D21", "D28", "D36", "D42")) # Set the level order of values in "Day" column
levels(sample_data(flu.sam)$Day) #Set the level order of values in "Day" column to D0"  "D1"  "D3"  "D7"  "D10" "D14" "D21" "D28" "D36" "D42"   
    
#Calculate the average shannon, invsimpson, numOTUs for each "All" subtype within flu.sam
average.shannon.invsimpson.numOTUs <- aggregate(flu.sam[, 6:8], list(flu.sam$All), mean)
print(average.shannon.invsimpson.numOTUs)
#Output:
#       Group.1   numOTUS  shannon invsimpson
#1   D0 Control  69.10000 2.565335   6.377505
#2       D0 IAV  51.20000 2.169701   4.399913
#3   D1 Control  50.90000 2.253141   5.126906
#4       D1 IAV  45.20000 2.030144   4.128696
#5  D10 Control  62.70000 2.445356   5.112930
#6      D10 IAV  51.70000 2.305930   5.136920
#7  D14 Control  81.10000 2.853185   7.046119
#8      D14 IAV  92.20000 3.069216   8.757443
#9  D21 Control  79.90000 2.925234   7.111600
#10     D21 IAV  91.33333 3.135175   8.865022
#11 D28 Control 105.50000 3.511063  13.981237
#12     D28 IAV 103.20000 3.608026  15.135310
#13  D3 Control  85.60000 2.902182   7.333515
#14      D3 IAV  68.00000 2.571588   5.847328
#15 D36 Control  91.80000 3.152522  13.080790
#16     D36 IAV 109.11111 3.409859  15.417720
#17 D42 Control 108.40000 3.515769  17.795337
#18     D42 IAV 113.77778 3.477660  16.745868
#19  D7 Control  56.80000 2.382253   5.533712
#20      D7 IAV  26.20000 1.817367   4.238963
write.csv(average.shannon.invsimpson.numOTUs, file="Flu.average.shannon.invsimpson.num.OTUs.txt", row.names=TRUE)
flu.pairwise.wilcox.shannon.test <- pairwise.wilcox.test(flu.sam$shannon, flu.sam$All, p.adjust.method = 'none') #Calculate pairwise comparisons by "All" column of the shannon indices in "Shannon" column
print(flu.pairwise.wilcox.shannon.test) #Look at the results of 'Flu.pairwise.wilcox.shannon.test'
flu.pairwise.wilcox.invsimpson.test <- pairwise.wilcox.test(flu.sam$invsimpson, flu.sam$All, p.adjust.method = 'none') #Calculate pairwise comparisons by "All" column of the inverse simpson indices in "invsimpson" column
print(flu.pairwise.wilcox.invsimpson.test)
    
#Generate a box and whisker plot of shannon
flu.shannon <- ggplot(data = flu.sam, aes(x=Treatment, y=shannon, group=All, fill=Treatment)) +
  geom_boxplot(position = position_dodge2(preserve = 'total')) +
  facet_wrap(~Day, scales = 'free') +
  scale_y_continuous(name = "Shannon diversity") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
        strip.text.x = element_text(size=10),
        axis.title.y = element_text(size=15))
# "free" within "facet_wrap" allows each plot to customize the scale to the specific data set (no forced scaling applied to all plots)
# "position = position_dodge2(preserve = 'total')" fixes the ggplot box width, making them wider, prevents narrow boxes from forming in the plot
flu.shannon
    
#Generate a box and whisker plot of inverse simpson diversity
flu.invsimp <- ggplot(data = flu.sam, aes(x=Treatment, y=invsimpson, group=All, fill=Treatment)) +
  geom_boxplot(position = position_dodge2(preserve = 'total')) +
  facet_wrap(~Day, scales = 'free') +
  scale_y_continuous(name = "Inverse Simpson diversity") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size = 10, angle = 45, hjust = 1),
        strip.text.x = element_text(size=10),
        axis.title.y = element_text(size=15))
flu.invsimp
    
#Combining plots
flu.combalpha <- plot_grid(flu.shannon + theme(legend.position = "none"), flu.invsimp, labels = "AUTO")
flu.combalpha

#Save 'flu.combalpha' as a .tiff for publication, 500dpi
ggsave("Figure_4.tiff", plot=flu.combalpha, width = 10, height = 5, dpi = 500, units =c("in"))
```
</details>

***
# Magnitude of Change in Nasal Microbiota
**Goal**: Plot the F-statistic from PERMANOVA pairwise comparisons of Control and IAV groups, over time (displays the magnitude of change in the nasal bacterial community structure of the IAV group relative to control)

1. Files needed:
  - Flu_MagnitudeOfChange.csv
    - Flu_MagnitudeOfChange.csv was created from Flu.pairwisecomparisons.csv
    - To make the Flu_MagnitudeOfChange.csv file, open Flu.pairwisecomparisons.csv file in excel, copy columns "F. Model" through "p.adjusted2" and paste in a separate spreadsheet. Add "Day" and "Treatment" columns and save as "Flu_MagnitudeOfChange.csv".
2. The F-values were obtained from 'Flu.pairwisecomparisons.csv' data generated in the "Beta diversity" section

<details><summary>SRD129_NasalMagnitudeOfChange_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load library packages
library(vegan)
library(ggplot2)
library(dplyr)
library(tidyr)
library(tidyverse)
library(phyloseq)
library(RColorBrewer)
library(philentropy)
library(cowplot)
library(ggrepel)

########################

flu.mag <- read.csv("./data/Flu_MagnitudeOfChange.csv")
class(flu.mag)
flu.mag$Day <- factor(flu.mag$Day) #Encode "Day" as a factor
flu.mag2 <- ggplot(data=flu.mag, aes(x=Day, y=F.Model, group=Treatment)) +
      geom_line(aes(color=Treatment)) + geom_point(aes(color=Treatment)) +
      ylab("PERMANOVA F vs control \n(difference relative to control)") +
      scale_fill_manual(values=c("#00BFC4")) +
      scale_color_manual(values=c("#00BFC4")) + 
      geom_label_repel(aes(label=p.adjusted2), box.padding = 0.35, point.padding=0.5,segment.color = 'grey50') +
      theme_classic(base_size = 12) +
      theme(axis.text.y = element_text(size = 14), axis.text.x = element_text(size=14), axis.title.x = element_text(size=14), axis.title.y = element_text(size=14), legend.text=element_text(size=14), legend.title=element_text(size=14)) +
      labs(color="Treatment group")
flu.mag2
    
#Save 'flu.mag2' as a .tiff for publication, 500dpi
ggsave("Figure_3.tiff", plot=flu.mag2, width = 10, height = 5, dpi = 500, units =c("in"))
```
</details>

***
# Differential Abundance of Genera in Nasal Microbiota using DESeq2
**Goal**: Use DESeq2 package to identify nasal microbial genera that were differentially abundant between treatment groups and control group

1. Files needed:
  - Mothur shared file: SRD129Flu.outsingletons.abund.opti_mcc.shared
  - Mothur constaxonomy file: SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
  - Metadata: SRD129metadata.csv

<details><summary>SRD129_Nasal_DeSeq2_genus_Flu.R</summary>
````{r echo=TRUE, eval=FALSE}
#Load library packages
library(DESeq2)
library(phyloseq)
library(ggplot2)
library(tidyr)
library("wesanderson")
library(plotly)
library(gapminder)
library(cowplot)
    
#Annotation
#IC = IAV, control
    
#Preparing objects for DESeq2: load files
otu2 <- import_mothur(mothur_shared_file = './data/SRD129Flu.outsingletons.abund.opti_mcc.shared')
taxo2 <- import_mothur(mothur_constaxonomy_file = './data/SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy')
taxo2
meta2 <- read.table(file = './data/SRD129metadata.csv', sep = ',', header = TRUE)

#Organize meta file
rownames(meta2) <- meta2$Sample
meta2 <- meta2[,-1] #Remove Sample column
meta2$Set <- paste(meta2$Day, meta2$Treatment, sep = '_') #Create 'Set' column that combines Day and Treatment

#Make phyloseq object SRD129 (combine taxonomy, OTU, and metadata)
phy_meta2 <- sample_data(meta2) 
SRD129 <- phyloseq(otu2, taxo2)
SRD129 <- merge_phyloseq(SRD129, phy_meta2)   #Combines the metadata with this phyloseq object
colnames(tax_table(SRD129))
colnames(tax_table(SRD129)) <- c('Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus')
SRD129

#Prune and subset by genus rank
SRD129 <- prune_samples(sample_sums(SRD129) > 2000, SRD129)  #This removes samples that have fewer than 2000 sequences associated with them.
SRD129 <- prune_taxa(taxa_sums(SRD129) > 10, SRD129)        #Removes OTUs that occur less than 10 times globally
tax_table(SRD129) [1:5, 1:6]  #See what's in tax_table first 5 rows, first 6 columns
SRD129.genus <- tax_glom(SRD129, taxrank = "Genus")
#This method merges species that have the same taxonomy at a certain taxanomic rank. 
#Its approach is analogous to tip_glom, but uses categorical data instead of a tree. 



# PRIMARY COMPARISONS TO MAKE #

# NMDS plot showed that dispersion is different between days, so I subsetted by day

# Important comparisons to make (significant changes in beta diversity between treatments): Compare Days 14, 21, 28 IAV and Control

# Other comparisons to make (no significant changes in beta diversity between treatments): Compare Days 0, 1, 3, 7, 10, 36, 42 IAV and Control


## Day 0 Nasal #########################################################

sample_data(SRD129.genus)
SRD129.D0 <- subset_samples(SRD129.genus, Day == 'D0')
sample_sums(SRD129.D0)
colnames(otu_table(SRD129.D0)) #Check on all the sample names
SRD129.D0 <- prune_taxa(taxa_sums(SRD129.D0) > 1, SRD129.D0)
#If taxa_sums is >1, then it will print that out in SRD129.D0 object and not include anything with <1.
rowSums(SRD129.D0@otu_table)
SRD129.D0.De <- phyloseq_to_deseq2(SRD129.D0, ~ Set)
# ~Set: whatever you want to group data by and use to designate ellipses with
SRD129.D0.De <- DESeq(SRD129.D0.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using "meta2" dataframe): 
sum(meta2$Set == "D0_IAV")
#IAV = 10
sum(meta2$Set == "D0_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D0.De$Set
res.D0.ic = results(SRD129.D0.De, contrast=c("Set","D0_IAV","D0_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D0.ic = res.D0.ic[which(res.D0.ic$padj < .05), ]
sigtab.D0.ic = cbind(as(sigtab.D0.ic, "data.frame"), as(tax_table(SRD129.D0)[rownames(sigtab.D0.ic), ], "matrix"))
format(sigtab.D0.ic$padj, scientific = TRUE)
sigtab.D0.ic$newp <- format(round(sigtab.D0.ic$padj, digits = 3), scientific = TRUE)
sigtab.D0.ic$Treatment <- ifelse(sigtab.D0.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D0.ic
sum.sigtab.D0.ic <- summary(sigtab.D0.ic)
sum.sigtab.D0.ic

#ggplot
deseq.D0.ic <- ggplot(sigtab.D0.ic, aes(x=reorder(rownames(sigtab.D0.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D0.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 0')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control="#999999", IAV = "#CC0066"))
deseq.D0.ic

#Add OTU and comparisons columns
sigtab.D0.ic
sigtab.D0.ic$OTU <- rownames(sigtab.D0.ic)
sigtab.D0.ic
sigtab.D0.ic$comp <- 'D0_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(sigtab.D0.ic)

## Day 1 Nasal ######################################################################

sample_data(SRD129.genus)
SRD129.D1 <- subset_samples(SRD129.genus, Day == 'D1')
sample_sums(SRD129.D1)
colnames(otu_table(SRD129.D1)) #check on all the sample names
SRD129.D1 <- prune_taxa(taxa_sums(SRD129.D1) > 1, SRD129.D1)
rowSums(SRD129.D1@otu_table)
SRD129.D1.De <- phyloseq_to_deseq2(SRD129.D1, ~ Set)
SRD129.D1.De <- DESeq(SRD129.D1.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D1_IAV")
#IAV = 10
sum(meta2$Set == "D1_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D1.De$Set
res.D1.ic = results(SRD129.D1.De, contrast=c("Set","D1_IAV","D1_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D1.ic = res.D1.ic[which(res.D1.ic$padj < .05), ]
sigtab.D1.ic = cbind(as(sigtab.D1.ic, "data.frame"), as(tax_table(SRD129.D1)[rownames(sigtab.D1.ic), ], "matrix"))
format(sigtab.D1.ic$padj, scientific = TRUE)
sigtab.D1.ic$newp <- format(round(sigtab.D1.ic$padj, digits = 3), scientific = TRUE)
sigtab.D1.ic$Treatment <- ifelse(sigtab.D1.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D1.ic
sum.sigtab.D1.ic <- summary(sigtab.D1.ic)
sum.sigtab.D1.ic

#ggplot
deseq.D1.ic <- ggplot(sigtab.D1.ic, aes(x=reorder(rownames(sigtab.D1.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D1.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 1')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D1.ic

#Add OTU and comparisons columns
sigtab.D1.ic
sigtab.D1.ic$OTU <- rownames(sigtab.D1.ic)
sigtab.D1.ic
sigtab.D1.ic$comp <- 'D1_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D1.ic)

## Day 3 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D3 <- subset_samples(SRD129.genus, Day == 'D3')
sample_sums(SRD129.D3)
colnames(otu_table(SRD129.D3)) #check on all the sample names
SRD129.D3 <- prune_taxa(taxa_sums(SRD129.D3) > 1, SRD129.D3)
rowSums(SRD129.D3@otu_table)
SRD129.D3.De <- phyloseq_to_deseq2(SRD129.D3, ~ Set)
SRD129.D3.De <- DESeq(SRD129.D3.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D3_IAV")
#IAV = 10
sum(meta2$Set == "D3_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D3.De$Set
res.D3.ic = results(SRD129.D3.De, contrast=c("Set","D3_IAV","D3_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D3.ic = res.D3.ic[which(res.D3.ic$padj < .05), ]
sigtab.D3.ic = cbind(as(sigtab.D3.ic, "data.frame"), as(tax_table(SRD129.D3)[rownames(sigtab.D3.ic), ], "matrix"))
format(sigtab.D3.ic$padj, scientific = TRUE)
sigtab.D3.ic$newp <- format(round(sigtab.D3.ic$padj, digits = 3), scientific = TRUE)
sigtab.D3.ic$Treatment <- ifelse(sigtab.D3.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D3.ic
sum.sigtab.D3.ic <- summary(sigtab.D3.ic)
sum.sigtab.D3.ic

#ggplot
deseq.D3.ic <- ggplot(sigtab.D3.ic, aes(x=reorder(rownames(sigtab.D3.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D3.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 3')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D3.ic

#Add OTU and comparisons columns
sigtab.D3.ic
sigtab.D3.ic$OTU <- rownames(sigtab.D3.ic)
sigtab.D3.ic
sigtab.D3.ic$comp <- 'D3_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D3.ic)

## Day 7 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D7 <- subset_samples(SRD129.genus, Day == 'D7')
sample_sums(SRD129.D7)
colnames(otu_table(SRD129.D7)) #check on all the sample names
SRD129.D7 <- prune_taxa(taxa_sums(SRD129.D7) > 1, SRD129.D7)
rowSums(SRD129.D7@otu_table)
SRD129.D7.De <- phyloseq_to_deseq2(SRD129.D7, ~ Set)
SRD129.D7.De <- DESeq(SRD129.D7.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D7_IAV")
#IAV = 10
sum(meta2$Set == "D7_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D7.De$Set
res.D7.ic = results(SRD129.D7.De, contrast=c("Set","D7_IAV","D7_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D7.ic = res.D7.ic[which(res.D7.ic$padj < .05), ]
sigtab.D7.ic = cbind(as(sigtab.D7.ic, "data.frame"), as(tax_table(SRD129.D7)[rownames(sigtab.D7.ic), ], "matrix"))
format(sigtab.D7.ic$padj, scientific = TRUE)
sigtab.D7.ic$newp <- format(round(sigtab.D7.ic$padj, digits = 3), scientific = TRUE)
sigtab.D7.ic$Treatment <- ifelse(sigtab.D7.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D7.ic
sum.sigtab.D7.ic <- summary(sigtab.D7.ic)
sum.sigtab.D7.ic

#ggplot
deseq.D7.ic <- ggplot(sigtab.D7.ic, aes(x=reorder(rownames(sigtab.D7.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D7.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 7')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D7.ic

#Add OTU and comparisons columns
sigtab.D7.ic
sigtab.D7.ic$OTU <- rownames(sigtab.D7.ic)
sigtab.D7.ic
sigtab.D7.ic$comp <- 'D7_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D7.ic)

## Day 10 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D10 <- subset_samples(SRD129.genus, Day == 'D10')
sample_sums(SRD129.D10)
colnames(otu_table(SRD129.D10)) #check on all the sample names
SRD129.D10 <- prune_taxa(taxa_sums(SRD129.D10) > 1, SRD129.D10)
rowSums(SRD129.D10@otu_table)
SRD129.D10.De <- phyloseq_to_deseq2(SRD129.D10, ~ Set)
SRD129.D10.De <- DESeq(SRD129.D10.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D10_IAV")
#IAV = 10
sum(meta2$Set == "D10_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D10.De$Set
res.D10.ic = results(SRD129.D10.De, contrast=c("Set","D10_IAV","D10_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D10.ic = res.D10.ic[which(res.D10.ic$padj < .05), ]
sigtab.D10.ic = cbind(as(sigtab.D10.ic, "data.frame"), as(tax_table(SRD129.D10)[rownames(sigtab.D10.ic), ], "matrix"))
format(sigtab.D10.ic$padj, scientific = TRUE)
sigtab.D10.ic$newp <- format(round(sigtab.D10.ic$padj, digits = 3), scientific = TRUE)
sigtab.D10.ic$Treatment <- ifelse(sigtab.D10.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D10.ic
sum.sigtab.D10.ic <- summary(sigtab.D10.ic)
sum.sigtab.D10.ic

#ggplot
deseq.D10.ic <- ggplot(sigtab.D10.ic, aes(x=reorder(rownames(sigtab.D10.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D10.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 10')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D10.ic

#Add OTU and comparisons columns
sigtab.D10.ic
sigtab.D10.ic$OTU <- rownames(sigtab.D10.ic)
sigtab.D10.ic
sigtab.D10.ic$comp <- 'D10_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D10.ic)

## Day 14 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D14 <- subset_samples(SRD129.genus, Day == 'D14')
sample_sums(SRD129.D14)
colnames(otu_table(SRD129.D14)) #check on all the sample names
SRD129.D14 <- prune_taxa(taxa_sums(SRD129.D14) > 1, SRD129.D14)
rowSums(SRD129.D14@otu_table)
SRD129.D14.De <- phyloseq_to_deseq2(SRD129.D14, ~ Set)
SRD129.D14.De <- DESeq(SRD129.D14.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D14_IAV")
#IAV = 10
sum(meta2$Set == "D14_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D14.De$Set
res.D14.ic = results(SRD129.D14.De, contrast=c("Set","D14_IAV","D14_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D14.ic = res.D14.ic[which(res.D14.ic$padj < .05), ]
sigtab.D14.ic = cbind(as(sigtab.D14.ic, "data.frame"), as(tax_table(SRD129.D14)[rownames(sigtab.D14.ic), ], "matrix"))
format(sigtab.D14.ic$padj, scientific = TRUE)
sigtab.D14.ic$newp <- format(round(sigtab.D14.ic$padj, digits = 3), scientific = TRUE)
sigtab.D14.ic$Treatment <- ifelse(sigtab.D14.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D14.ic
sum.sigtab.D14.ic <- summary(sigtab.D14.ic)
sum.sigtab.D14.ic

#ggplot
deseq.D14.ic <- ggplot(sigtab.D14.ic, aes(x=reorder(rownames(sigtab.D14.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D14.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 14')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D14.ic

#Add OTU and comparisons columns
sigtab.D14.ic
sigtab.D14.ic$OTU <- rownames(sigtab.D14.ic)
sigtab.D14.ic
sigtab.D14.ic$comp <- 'D14_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D14.ic)

## Day 21 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D21 <- subset_samples(SRD129.genus, Day == 'D21')
sample_sums(SRD129.D21)
colnames(otu_table(SRD129.D21))
SRD129.D21 <- prune_taxa(taxa_sums(SRD129.D21) > 1, SRD129.D21)
rowSums(SRD129.D21@otu_table)
SRD129.D21.De <- phyloseq_to_deseq2(SRD129.D21, ~ Set)
SRD129.D21.De <- DESeq(SRD129.D21.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D21_IAV")
#IAV = 9
sum(meta2$Set == "D21_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D21.De$Set
res.D21.ic = results(SRD129.D21.De, contrast=c("Set","D21_IAV","D21_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D21.ic = res.D21.ic[which(res.D21.ic$padj < .05), ]
sigtab.D21.ic = cbind(as(sigtab.D21.ic, "data.frame"), as(tax_table(SRD129.D21)[rownames(sigtab.D21.ic), ], "matrix"))
format(sigtab.D21.ic$padj, scientific = TRUE)
sigtab.D21.ic$newp <- format(round(sigtab.D21.ic$padj, digits = 3), scientific = TRUE)
sigtab.D21.ic$Treatment <- ifelse(sigtab.D21.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D21.ic
sum.sigtab.D21.ic <- summary(sigtab.D21.ic)
sum.sigtab.D21.ic

#ggplot
deseq.D21.ic <- ggplot(sigtab.D21.ic, aes(x=reorder(rownames(sigtab.D21.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D21.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 21')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D21.ic

#Add OTU and comparisons columns
sigtab.D21.ic
sigtab.D21.ic$OTU <- rownames(sigtab.D21.ic)
sigtab.D21.ic
sigtab.D21.ic$comp <- 'D21_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D21.ic)

## Day 28 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D28 <- subset_samples(SRD129.genus, Day == 'D28')
sample_sums(SRD129.D28)
colnames(otu_table(SRD129.D28))
SRD129.D28 <- prune_taxa(taxa_sums(SRD129.D28) > 1, SRD129.D28)
rowSums(SRD129.D28@otu_table)
SRD129.D28.De <- phyloseq_to_deseq2(SRD129.D28, ~ Set)
SRD129.D28.De <- DESeq(SRD129.D28.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D28_IAV")
#IAV = 5
sum(meta2$Set == "D28_Control")
#Control = 4

#Extract results from a DESeq analysis, organize table
SRD129.D28.De$Set
res.D28.ic = results(SRD129.D28.De, contrast=c("Set","D28_IAV","D28_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D28.ic = res.D28.ic[which(res.D28.ic$padj < .05), ]
sigtab.D28.ic = cbind(as(sigtab.D28.ic, "data.frame"), as(tax_table(SRD129.D28)[rownames(sigtab.D28.ic), ], "matrix"))
format(sigtab.D28.ic$padj, scientific = TRUE)
sigtab.D28.ic$newp <- format(round(sigtab.D28.ic$padj, digits = 3), scientific = TRUE)
sigtab.D28.ic$Treatment <- ifelse(sigtab.D28.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D28.ic
sum.sigtab.D28.ic <- summary(sigtab.D28.ic)
sum.sigtab.D28.ic

#ggplot
deseq.D28.ic <- ggplot(sigtab.D28.ic, aes(x=reorder(rownames(sigtab.D28.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D28.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 28')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D28.ic

#Add OTU and comparisons columns
sigtab.D28.ic
sigtab.D28.ic$OTU <- rownames(sigtab.D28.ic)
sigtab.D28.ic
sigtab.D28.ic$comp <- 'D28_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D28.ic)

## Day 36 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D36 <- subset_samples(SRD129.genus, Day == 'D36')
sample_sums(SRD129.D36)
colnames(otu_table(SRD129.D36))
SRD129.D36 <- prune_taxa(taxa_sums(SRD129.D36) > 1, SRD129.D36)
rowSums(SRD129.D36@otu_table)
SRD129.D36.De <- phyloseq_to_deseq2(SRD129.D36, ~ Set)
SRD129.D36.De <- DESeq(SRD129.D36.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D36_IAV")
#IAV = 9
sum(meta2$Set == "D36_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D36.De$Set
res.D36.ic = results(SRD129.D36.De, contrast=c("Set","D36_IAV","D36_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D36.ic = res.D36.ic[which(res.D36.ic$padj < .05), ]
sigtab.D36.ic = cbind(as(sigtab.D36.ic, "data.frame"), as(tax_table(SRD129.D36)[rownames(sigtab.D36.ic), ], "matrix"))
format(sigtab.D36.ic$padj, scientific = TRUE)
sigtab.D36.ic$newp <- format(round(sigtab.D36.ic$padj, digits = 3), scientific = TRUE)
sigtab.D36.ic$Treatment <- ifelse(sigtab.D36.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D36.ic
sum.sigtab.D36.ic <- summary(sigtab.D36.ic)
sum.sigtab.D36.ic

#ggplot
deseq.D36.ic <- ggplot(sigtab.D36.ic, aes(x=reorder(rownames(sigtab.D36.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D36.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 36')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D36.ic

#Add OTU and comparisons columns
sigtab.D36.ic
sigtab.D36.ic$OTU <- rownames(sigtab.D36.ic)
sigtab.D36.ic
sigtab.D36.ic$comp <- 'D36_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D36.ic)

## Day 42 Nasal ############################################################

sample_data(SRD129.genus)
SRD129.D42 <- subset_samples(SRD129.genus, Day == 'D42')
sample_sums(SRD129.D42)
colnames(otu_table(SRD129.D42))
SRD129.D42 <- prune_taxa(taxa_sums(SRD129.D42) > 1, SRD129.D42)
rowSums(SRD129.D42@otu_table)
SRD129.D42.De <- phyloseq_to_deseq2(SRD129.D42, ~ Set)
SRD129.D42.De <- DESeq(SRD129.D42.De, test = "Wald", fitType = "parametric")

meta2$Set
#Number of pigs per group (using meta2 dataframe): 
sum(meta2$Set == "D42_IAV")
#IAV = 9
sum(meta2$Set == "D42_Control")
#Control = 10

#Extract results from a DESeq analysis, organize table
SRD129.D42.De$Set
res.D42.ic = results(SRD129.D42.De, contrast=c("Set","D42_IAV","D42_Control"),cooksCutoff = FALSE, pAdjustMethod = 'BH')
sigtab.D42.ic = res.D42.ic[which(res.D42.ic$padj < .05), ]
sigtab.D42.ic = cbind(as(sigtab.D42.ic, "data.frame"), as(tax_table(SRD129.D42)[rownames(sigtab.D42.ic), ], "matrix"))
format(sigtab.D42.ic$padj, scientific = TRUE)
sigtab.D42.ic$newp <- format(round(sigtab.D42.ic$padj, digits = 3), scientific = TRUE)
sigtab.D42.ic$Treatment <- ifelse(sigtab.D42.ic$log2FoldChange >=0, "IAV", "Control")

#Summarize sigtab.D42.ic
sum.sigtab.D42.ic <- summary(sigtab.D42.ic)
sum.sigtab.D42.ic

#ggplot
deseq.D42.ic <- ggplot(sigtab.D42.ic, aes(x=reorder(rownames(sigtab.D42.ic), log2FoldChange), y=log2FoldChange, fill = Treatment)) +
  geom_bar(stat='identity') + geom_text(aes(x=rownames(sigtab.D42.ic), y=0, label = paste(Family,Genus, sep = ' ')), size=5)+ labs(x="Family genus")+
  theme(axis.text.x=element_text(color = 'black', size = 13),
        axis.text.y=element_text(color = 'black', size=13, face = 'italic'), 
        axis.title.x=element_text(size = 15),
        axis.title.y=element_text(size = 15))+ ggtitle('Differentially Abundant OTUs in Influenza A Virus Group Relative to Control at Nasal Site on Day 42')+ coord_flip() +
  theme(plot.title = element_text(size = 18, hjust=0.5), legend.text = element_text(size=12), legend.title = element_text(size=13)) +
  scale_fill_manual(values = c(Control = "#999999", IAV = "#CC0066"))
deseq.D42.ic

#Add OTU and comparisons columns
sigtab.D42.ic
sigtab.D42.ic$OTU <- rownames(sigtab.D42.ic)
sigtab.D42.ic
sigtab.D42.ic$comp <- 'D42_IAVvsControl'

#Create final significant comparisons table
final.sigtab <- rbind(final.sigtab, sigtab.D42.ic)

## write csv ########################################
write.csv(final.sigtab, file= "SRD129_FinalDiffAbundGenus.csv")


## Plots of Differentially Abundant Nasal Genera Combined for Each Pairwise Comparison of IAV and Control Groups ########
library("ggsci")

#IAV and Control Log2fold Plot Part A
final_ic <- sigtab.D0.ic
final_ic <- rbind(final_ic, sigtab.D1.ic, sigtab.D3.ic, sigtab.D7.ic, sigtab.D10.ic, 
                  sigtab.D14.ic, sigtab.D21.ic, sigtab.D28.ic, sigtab.D36.ic, sigtab.D42.ic)
final_ic$Family_Genus <- paste(final_ic$Family, final_ic$Genus) #create new column with Family_Genus
final_ic$comp
class(final_ic)
final_ic$comp <- factor(final_ic$comp, levels=c("D0_IAVvsControl",
                                                "D1_IAVvsControl", "D3_IAVvsControl", "D7_IAVvsControl",
                                                "D10_IAVvsControl", "D14_IAVvsControl", "D21_IAVvsControl",
                                                "D28_IAVvsControl", "D36_IAVvsControl", "D42_IAVvsControl"))
levels(final_ic$comp)
ic_plot <- ggplot(final_ic, aes(x=Family_Genus, log2FoldChange, fill = comp)) +
  geom_bar(stat='identity') +
  labs(x="Family Genus", y = "Total log2 Fold Change") +
  theme(axis.text.x=element_text(color = 'black', size = 18),
        axis.text.y=element_text(color = 'black', size=15), 
        axis.title.x=element_text(size = 20),
        axis.title.y=element_text(size = 20))+ 
  coord_flip() +
  scale_fill_igv(name = "comp") +
  ggtitle('Differentially Abundant Nasal Families and Genera between Influenza A Virus and Control Groups') + 
  theme(plot.title = element_text(size = 20), legend.text = element_text(size=20), legend.title = element_text(size=20))
ic_plot <- ic_plot + guides(fill=guide_legend(title="Day and treatment group"))
ic_plot

write.csv(final_ic, file= "SRD129_FinalDiffAbundNasalGenus_IAVcontrol.csv")

#Modified SRD129_FinalDiffAbundNasalGenus_IAVcontrol.csv in a spreadsheet editor by removing all genera except for Actinobacillus, Moraxella, Neisseriaceae_unclassified, Prevotellaceae_NK3B31_group,
#Prevotellaceae_unclassified, Staphylococcus, Streptococcus.
#Saved modified file as SRD129_FinalDiffAbundNasalGenus_IAVcontrolSelectList.csv

#IAV and control Log2fold Plot Part B
ic <- read.csv('SRD129_FinalDiffAbundNasalGenus_IAVcontrolSelectList.csv', header = TRUE, sep = ",")
head(ic[,1:10])
colnames(ic)
ic$DayComp <- sub('_[A-Za-z]+', '\\2', ic$comp)
unique(ic$DayComp)
ic$Day <- sub('_[A-Za-z]+', '\\2', ic$DayComp)
unique(ic$Day) #"D3"  "D7"  "D10" "D14" "D21" "D28"
ic$Day = factor(ic$Day, levels=c("D3","D7", "D10","D14", "D21", "D28"))
levels(ic$Day) #"D3"  "D7"  "D10" "D14" "D21" "D28"
(ic_logfoldplot <- ggplot(data=ic, aes(x=Day, y=log2FoldChange, fill=Treatment)) +
    geom_bar(stat = 'identity', position="dodge") +
    facet_wrap(~Genus, ncol = 3, scales = "free") + ylab('log2-fold change') +
    theme_gray()+
    theme(plot.title = element_text(hjust = 3)) +
    theme(axis.line = element_line()) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size=13),
          axis.text.y = element_text(size=13), 
          axis.title.x = element_text(size=15), 
          axis.title.y = element_text(size=15),
          legend.text=element_text(size=15), 
          legend.title=element_text(size=15)) +
    scale_fill_manual(values = c(Control = "#F8766D", IAV = "#00BFC4")))
ic_logfoldplot <- ic_logfoldplot + theme(strip.text = element_text(size= 13, face='italic'))
ic_logfoldplot

#Save 'ic_logfoldplot' as a .tiff for publication, 500dpi
ggsave("Figure_5.tiff", plot=ic_logfoldplot, width = 15, height = 7, dpi = 500, units =c("in"))
```
</details>

# Nasal Microbiota: Genus Abundance
**Goal**: Generate a list of percent total genera found in each treatment group per day for each tissue and creates a bar graph plot of the data 

1. Files needed:
  - SRD129Flu.outsingletons.abund.opti_mcc.shared
  - SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy
  - SRD129metadata.csv

<details><summary>SRD129_Genus_Flu.R</summary>
```{r echo=TRUE, eval=FALSE}
#Load libraries
library(phyloseq)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(cowplot)
library("ggsci")

#################

#Import files
otu <- import_mothur(mothur_shared_file = './data/SRD129Flu.outsingletons.abund.opti_mcc.shared')
taxo <- import_mothur(mothur_constaxonomy_file = './data/SRD129Flu.outsingletons.abund.opti_mcc.0.03.cons.taxonomy')
meta <- read.table('./data/SRD129metadata.csv', header = TRUE, sep = ",")
head(meta)
colnames(meta)[1] <- 'group' #Rename first column of "meta" as "group" temporarily. Will use "group" to set as rownames later and remove the "group" column
meta$Day<- gsub("D", "", meta$Day) #Remove "D"
meta$group <- as.character(meta$group)
head(meta)
phy_meta <- sample_data(meta) 
rownames(phy_meta) <- phy_meta$group
head(phy_meta)
phy_meta <- phy_meta[,-1]
head(phy_meta)

#Create phyloseq-class objects with "otu" and "taxo"
IAV <- phyloseq(otu, taxo)
IAV <- merge_phyloseq(IAV, phy_meta)  #This combines the 'phy_meta' metadata with 'IAV' phyloseq object
colnames(tax_table(IAV)) <- c('Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus')
sample_sums(IAV) #Calculate the sum of all OTUs for each sample,
IAV <- prune_taxa(taxa_sums(IAV) > 2, IAV)  #Removes OTUs that occur less than 2 times globally
IAV.genus <- tax_glom(IAV, 'Genus')
phyla_tab <- as.data.frame(t(IAV.genus@otu_table)) #Transpose 'IAV.genus' by "otu_table"
head(phyla_tab)
IAV.genus@tax_table[,6]
colnames(phyla_tab) <- IAV.genus@tax_table[,6] #Replace column names in phyla_tab from Otuxxxx with Genus names
phyla_tab2 <- phyla_tab/rowSums(phyla_tab) #Calculate the proportion of specific phyla per phyla column in 'phyla_tab'
head(phyla_tab2)
phyla_tab2$group <- rownames(phyla_tab2) #Create new column called "group" in 'phyla_tab2' containing rownames
head(phyla_tab2)
fobar <- merge(meta, phyla_tab2, by = 'group') #Merge 'meta' with 'phyla_tab2' by "group"
head(fobar)
fobar.gather <- fobar %>% gather(Genus, value, -(group:Treatment))  #This converts 'fobar' to long-form dataframe. 
#This also created new columns "Genus", "value"; it added columns "group" through "Treatment" before "Genus" and "value"
head(fobar.gather)

#Check to see where the extra "group" column is and remove the column
which(colnames(phyla_tab2)=="group") #Results say column 405 is "group" column
phyla_tab3 <- phyla_tab2[,-405] #Drop the 405th column
phyla_tab4 <- phyla_tab3[,colSums(phyla_tab3)>0.1] #Keep the columns that have greater than 0.1 value
phyla_tab4$group <- rownames(phyla_tab4) #Rename rownames as "group"
fobar2 <- merge(meta, phyla_tab4, by = 'group')
head(fobar2)
fobar2.gather <- fobar2 %>% gather(Genus, value, -(group:Treatment)) 
head(fobar2.gather)

#Create "All" column with "Day" and "Treatment" in 'fobar2.gather'
fobar2.gather$All <- paste(fobar2.gather$Day, fobar2.gather$Treatment, sep = '_')

#Count the number of unique items in 'fobar2.gather'. We're interested in the total unique number of genera
fobar2.gather %>% summarise_each(funs(n_distinct)) #86 total unique genera
fobar2.gather <- fobar2.gather %>% group_by(All) %>% mutate(value2=(value/(length(All)/86))*100) #86 refers to number of Genera

#Subset each "All" group from fobar2.gather and save as individual csv files.
#Two examples:
D0_Control.genus <- subset(fobar2.gather, All=="0_Control")
write.csv(D0_Control.genus, file="D0_Control.genus.csv")

D1_IAV.genus <- subset(fobar2.gather, All=="D1_IAV.genus")
write.csv(D1_IAV.genus, file="D4_IAV.genus")

#Calculate the total % percent abundance of each genera on each sample (I used JMP to do this) 
#and save results in a spreadsheet editor such as Excel (see D0_Control.genus.xlsx for an example)
#Since we are only interested in genera that are above 2% abundance, 
#calculate total percentage of all other genera that are less than 2% in the spreadsheet and label as "Other".
#Create a new spreadsheet and label as "Nasal genus.csv". 
#Create the following columns: Day, Treatment group, Percent Abundance, and Genus. 
#Copy the list of genera and their percent abundances from each of the individual Excel files to the respective "Nasal genus.csv" spreadsheet.
#Fill in the other columns manually (Day, Treatment Group). 
#Save this as SRD129_Nasal_GenusPercentAbundance.csv. Continue to the next step.
nasalgen <- read.table('SRD129_Nasal_GenusPercentAbundance.csv', header = TRUE, sep = ",")
head(nasalgen)
unique(nasalgen.2$Day) #D0  D1  D3  D7  D10 D14 D21 D28 D36 D42
nasalgen.2$Day = factor(nasalgen.2$Day, levels = c("D0", "D1", "D3", "D7", "D10", "D14", "D21", "D28", "D36", "D42"))
nasalgen.2$More.than.2=as.character(nasalgen.2$Genus)
str(nasalgen.2$More.than.2) #Compactly display the internal structure of an R object,
nasalgen.2$More.than.2[nasalgen.2$Percent.abundance<1]<-"Other"
write.csv(nasalgen.2, file = "SRD129_Nasal_GenusPercentAbundanceAbove1percent.csv")

#To make sure the total percent abundance of all organisms for each day adds up to 100%, 
#modify the percent abundance for "Other" for each day in "SRD129_Nasal_GenusPercentAbundanceAbove1percent.csv" in a spreadsheet editor 
#and save as "SRD129_Nasal_GenusPercentAbundanceAbove1PercentAddTo100FINAL.csv"

#Create nasal genera plot
nasalgen.2 = read.csv("SRD129_Nasal_GenusPercentAbundanceAbove1PercentAddTo100FINAL.csv", header = TRUE)
levels(nasalgen.2$Day)  # "D0"  "D1"  "D10" "D14" "D21" "D28" "D3"  "D36" "D42" "D7" 
nasalgen.2$Day = factor(nasalgen.2$Day, levels = c("D0", "D1", "D3", "D7", "D10", "D14", "D21", "D28", "D36", "D42"))

#Nasalgen.2 abundance plot for each day, more than 1% genera
(nasalgen.2plot <- ggplot(data=nasalgen.2, aes(x=Treatment, y=Percent.abundance, fill=Genus)) +
    geom_bar(stat = 'identity') +
    #geom_bar(stat= 'identity', colour='black') +
    #theme(legend.key = element_rect = element_rect(colour='black', size=1.5)) +
    facet_grid(~Day) + ylab('Relative abundance (%) at nasal site') +
    theme(plot.title = element_text(hjust = 0.5)) +
    theme(axis.text.x=element_text(angle=45, hjust=1),
          axis.title.x = element_blank()) +
    scale_fill_igv(name = "Genus") +
    theme(legend.direction = "vertical") +
    theme(legend.text = element_text(face = 'italic')))

ggsave("FluControl_NasalGenera.tiff", plot=nasalgen.2plot, width = 15, height = 7, dpi = 500, units =c("in"))
```
</details>